### PR TITLE
add code guide for iOS

### PR DESCRIPTION
**Pull request by samuelbeek**
* Adds code guide for iOS.
* Links to our open source projects on iOS.
* Adds note on dependencies.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,8 +35,12 @@ Fetch the data and display it a grid like in the design:
 __Some points to pay attention to__
 * We don't expect you to just implement the design, we're also curious about your code.
 * You can go as far as you want with adding extra features.
+* If you use dependencies, please explain why.
 * Please WeTransfer your code to `mobile@wetransfer.com` or share it via a private repo on GitHub as we would like to re-use this assignment.
 * There is no perfect solution for this assignment. There are many ways to solve it.
+* For iOS:
+  * We have documented some code style guides [here](ios-guides/code-style.md).
+  * If you're interested in what our projects look like, we have two open source ones: [UI Notifications](https://github.com/wetransfer/UINotifications), [Mocker](https://github.com/WeTransfer/Mocker)
```
Comment (AvdLee) [inline README.md]: Really like this. This should give them an idea on which level of code we expect!

File: README.md [hunk 2]
```diff
@@ -35,8 +35,12 @@ Fetch the data and display it a grid like in the design:
 __Some points to pay attention to__
 * We don't expect you to just implement the design, we're also curious about your code.
 * You can go as far as you want with adding extra features.
+* If you use dependencies, please explain why.
 * Please WeTransfer your code to `mobile@wetransfer.com` or share it via a private repo on GitHub as we would like to re-use this assignment.
 * There is no perfect solution for this assignment. There are many ways to solve it.
+* For iOS:
+  * We have documented some code style guides [here](ios-guides/code-style.md).
```
Comment (AvdLee) [inline README.md]: I would not say `some`, these are our code guides, right? We stand for them.

`We have documented our code style guides [here](ios-guides/code-style.md).`

File: README.md [hunk 3]
```diff
@@ -35,8 +35,12 @@ Fetch the data and display it a grid like in the design:
 __Some points to pay attention to__
 * We don't expect you to just implement the design, we're also curious about your code.
 * You can go as far as you want with adding extra features.
```
Comment (AvdLee) [inline README.md]: This reminds me of the fact we cán (and maybe should) actually expect more than the basics.

File: ios-guides/code-style.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+* Make sure the grouped files are in sync with the folder structure.
+* Protocols should have clear names, it should be clear that they aren't classes/structs.
+* Extensions should have an `Extensions` postfix and should be located in the `Extensions` directory.  
+* Never use code that is especially designed for tests in the main target. Instead: make protocols or subclass classes in the test target.
+
+#### Failures/Errors
+
+* Use __Fatal Error__ when we want to return something, but can't (e.g: in guard statements).
+* Use __Assertion Failure__ a crucial error occurs inside a method.
+* Always add a message
```
Comment (AvdLee) [inline ios-guides/code-style.md]: I wonder if these guidelines are still up to date, do you know?

Comment (samuelbeek) [reply ios-guides/code-style.md]: they are

**Comment by samuelbeek** [comment]
@AvdLee they look up to date to me, what are you missing there?

File: README.md (hunk 3 above)
Comment (samuelbeek) [reply README.md]: No, we don't expect more. If they implement the solution perfectly according to our coding guidelines, then they're good right? @AvdLee

File: README.md (hunk 2 above)
Comment (samuelbeek) [reply README.md]: Good one

File: README.md (hunk 3 above)
Comment (AvdLee) [reply README.md]: I was more pointing to our submission so far. It's not that we didn't mention to implement more than requested.

You're right, it's not mandated.

File: ios-guides/code-style.md (hunk 4 above)
Comment (AvdLee) [reply ios-guides/code-style.md]: cool!

**Pull request merged**